### PR TITLE
golang dep support

### PR DESCRIPTION
**Pull request by caarlos0**
I don't do JS usually, so forgive me if it's not good :P 

Either way, I was able to get the dependencies and etc, but I still need to work on making the opencollective projects display properly.

Right now, it looks like this:

image

image

image

refs #75

### CODE REVIEW

**Comment by znarf** [comment]
Excellent!

The Open Collective matching is today a bit tricky, we're currently relying on a static list in https://github.com/opencollective/backyourstack/blob/master/src/data/projects.json

I have a script to help populate this file, but it's not ready to be shared in the repository yet.

Basically, we would need to iterate over all Open Collectives, then iterate over all their GitHub repositories, filter the one in the Go language, then detect the relevant package ids.

While I'm working on a version of this script that I can share in the repository, maybe we can identify a few Go projects that are on Open Collective and add them to the list manually.

https://opencollective.com/search?q=%20go
https://opencollective.com/search?q=%20golang

**Comment by caarlos0** [comment]
hmm, that leads me to another question: GoReleaser, for example, most of the time is not used as a lib. Projects using it usually have a `.goreleaser.yml` file in the root... do the project intent to support this kind of thing as well? There are probably several other projects that are not used as a lib but as a tool and have a config file somewhere inside the repository...

**Comment by znarf** [comment]
@caarlos0 I would like to focus on package managers on the short term.

That doesn't prevent to already start discussing new kind of strategies. And I do think It's a great idea to look at files at the root of the repository (like `.goreleaser.yml`) and make recommendations based on that. If implemented, I would like to make sure we preload properly the list of the files, make sure it's cheap for strategies to look at that data.

**Comment by znarf** [comment]
To keep you updated, I started looking at matching Go dependencies with Open Collectives projects but I still can't share anything today, maybe later this week!

**Comment by caarlos0** [comment]
Great, so, in the mean time, do you I need to change anything on this PR?

**Comment by znarf** [comment]
@caarlos0 The PR is looking great. However, we have another PR in line that's gonna conflict heavily with it. https://github.com/opencollective/backyourstack/pull/72 

My current plan is to merge #77 72 first, and #77 after. So, someone will have to resolve the conflicts (no problem for me if needed).

cc @marcandre

**Comment by caarlos0** [comment]
oh, no problem, I can deal with the merge conflicts as well :)

**Comment by marcandre** [comment]
I can resolve the conflicts too, no problem. If this PR is ready to be merged, it could be done first too and I'll rebase mine.

File: src/lib/dependencies/index.js [hunk 1]
```diff
@@ -30,6 +34,9 @@ function dependenciesStats (file) {
   if (file.type === 'nuget') {
     return nuget.dependenciesStats(file);
   }
+  if (file.type === 'dep') {
+    return nuget.dependenciesStats(file);
```
Comment (znarf) [inline src/lib/dependencies/index.js]: This should be `dep.dependenciesStats(file.text);`

File: src/lib/dependencies/dep.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    };
+  });
+  return Object.values(dependencies);
+}
+
+function isDependencyFile (file) {
+  return file.name === 'Gopkg.lock'
+}
+
+function detectProjectName () {
```
Comment (znarf) [inline src/lib/dependencies/dep.js]: Let's return `null` or `undefined` instead.

We'll change elsewhere the copy from "Unnamed project" to "Unknown project name"

**Comment by znarf** [comment]
Ok, let's merge this one first then. Just a few minor things to fix, see my comments. Also, run `npm run lint:fix` to correct a few warnings.

I added some Go projects to support in https://github.com/opencollective/backyourstack/commit/c11dee3ceedf648a7b0c3e30314dda23cb478070 We can test it for real now!

**Comment by caarlos0** [comment]
@znarf done! 🚀

File: src/lib/dependencies/index.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ function dependenciesStats (file) {
     return nuget.dependenciesStats(file);
   }
   if (file.type === 'dep') {
-    return nuget.dependenciesStats(file);
```
Comment (znarf) [inline src/lib/dependencies/index.js]: Still not there ;-) Should be `file.text` not `file`.

Comment (caarlos0) [reply src/lib/dependencies/index.js]: you're right, fixed

**Comment by marcandre** [comment]
Cool. Let me rebase and tweak #72 now :-)

**Pull request merged**